### DISCUSSION TITLE
Welcome to pccomponentes-buy-bot Discussions!

### DISCUSSION THREAD

**Discussion by elpatronaco** [elpatronaco/pccomponentes-buy-bot#3] [General]
ETELEEE me da palo escribir, pero aquí podéis hacer preguntas y hablar de lo que sea

**Comment by Rogue29x** [comment]
Bueno una 3060 Zotac no es tan mala, lo unico las temperaturas y los W pero dentro de lo que cabe, como bien dices no está el mercado para tirar cohetes. Tienen mas o menos próximamente alguna hora para ponerlas? o es aleatorio?

**Reply by elpatronaco** [reply]
t la puedo soltar, es la zotac twin edge OC de 489€, pero tienes que tener el bot buscandola ahi a muerte antes. Avisame si la quieres

**Reply by Rogue29x** [reply]
Pues me lo pienso y te digo, de todos modos tengo una pregunta. como haces para guardarlas?

**Reply by elpatronaco** [reply]
la compro y le doy a pago x transferencia, te la guardan unos días hasta que pagas, sino desaparece. Digo antes de que se la lleve un minero la guardo yo por si se la tengo que soltar a alguien

**Reply by Rogue29x** [reply]
Entonces, el pago por tarjeta tengo que quitarlo, para hacer eso. Las 580 salen? la verdad que para jugar al csgo y poco mas me sobra , tengo una 1070 pero me está fallando y como la economía no está muy bien en estos tiempos que digamos. Creo que con una 580 puedo ir tirando

**Reply by elpatronaco** [reply]
bua pues no tengo ni idea tio, no las he visto en stock porque tampoco he ido a por ellas. Aun así las veo demasiado caras por lo que valen. La 580 cuando salió costaba 220 por ahí, y ahora 300 napos

**Reply by elpatronaco** [reply]
lo del pago por transfer mi bot lo hace si no tienes ninguna tarjeta puesta en tu cuenta y no le has dado ninguna al instanciarlo, si le has puesto alguna tarjeta o tienes una guardada en tu cuenta tirará con eso

**Reply by GuabinaCore** [reply]
Buenas, una pregunta como haces para que busque varias tarjetas a la vez? abriendo el bot varias veces o pones varias urls en el "link"? Porque ahorita mismo es imposible esperar que aparezca mi favorita. Otra cosa @elpatronaco me sirve hasta la zotac que tienes guardada. ;)

**Reply by elpatronaco** [reply]
> Buenas, una pregunta como haces para que busque varias tarjetas a la vez? abriendo el bot varias veces o pones varias urls en el "link"? Porque ahorita mismo es imposible esperar que aparezca mi favorita. Otra cosa @elpatronaco me sirve hasta la zotac que tienes guardada. ;)

De momento el bot en sí solo puede buscar una, aunque estoy trabajando para que busque las que sean. Sobre la zotac avísame, cuando lo pongas a buscar la suelto y reza para que te la compre. Es [esta ](https://www.pccomponentes.com/zotac-geforce-rtx-3060ti-d6-twin-edge-oc-8gb-gddr6)

**Reply by GuabinaCore** [reply]
Eso seria genial @elpatronaco porque al abrir varias tarjetas el bot consume muchisimos recursos a mi parecer... Es zotac me viene bien, en este punto ya cualquiera me viene bien xD como hacemos para ponernos deacuerdo para que la sueltes? Podemos cuadrar una hora en especifico o algo? Saludos!

**Comment by Rogue29x** [comment]
Cuando el bot funciona pero te dice esto? 
[1060:19824:0127/134315.776:ERROR:ssl_client_socket_impl.cc(962)] handshake failed; returned -1, SSL error code 1, net_error -113

luego sigue con Product is not yet in stock (Wed, 27 Jan 2021 12:43:16 GMT)

**Reply by elpatronaco** [reply]
No tengo ni idea, no le hagas caso, selenium es muy random

**Reply by Rogue29x** [reply]
Vale, pues ahí está puesto a ver si sale algo. Lo dejáis 24h? o a ratos?

**Reply by elpatronaco** [reply]
yo cuando lo usé lo tenía 24/7 en una máquina virtual que hice en la nube, puedes buscar un tuto de un panchi de como hacer un windows server en Google Cloud gratis si no quieres tener encendido el ordenador siempre, pero es que no sabes cuando sacan los drops, a veces a las 5 de la mañana, a veces a las 3 de la tarde...

**Comment by Moiren1** [comment]
yo lo tengo con 10 tarjetas diferentes, a ver si tengo suerte y cazo 1, en una maquina virtual en mi servidor y se zampa el bot 4 threads de los 12 disponibles, el único fallo que veo, es que no siempre me refresca la pagina y cada x tiempo entro y le doy yo un click y ya vuelve a funcionar, pasa de manera aleatoria en algunas paginas, si a ti esto no te ha ocurrido entiendo que será por tener tantas paginas abiertas.

pd: yo tengo bloqueada una 3080 evga gaming ultra por 1065€, pero me parece carísimo. 

saludos.

**Reply by elpatronaco** [reply]
Sisi, traga CPU que da gusto, selenium está muy muy mal optimizado. Tengo la refactorización a puppeteer casi hecha y consume bastante menos, pero la acabaré el finde porque entre el curro y el insti no tengo tiempo pa nada mas. Y el tema que se te bloquea el bot creo que sé porque es, y a mí tambien me pasó. Cuando desconectas el RDP la sesión se pausa, no sé que hice pero busca "keep RDP session alive after disconnecting". Conseguí un script para desconectarme sin pausar la sesión, con lo que el bot seguía funcionando, pero ahora ns porqué no lo encuentro

**Comment by Rogue29x** [comment]


**Reply by Moiren1** [reply]
animo!!, hay mucha demanda y cada vez hay menos graficas y mas caras.

**Comment by Rogue29x** [comment]
@elpatronaco puedo hablarte por privador por favor? es una duda que tengo de como poner el pago por transferencia bancaria

**Comment by Noys33** [comment]
Hola, ante todo gracias. Tengo una duda, ¿que tengo que  hacer para que el bot compre con transferencia?
Si borro los valores de la tarjeta me da error.

**Comment by franVR10** [comment]
Buenas ! Soy nuevo en esto y se lo básico de lo básico. Estoy intentando conseguir una 3080 desesperadamente, por ello me encantaría usar este bot. Sin embargo, no me funciona, no llega ni a abrirme chrome. A ver si pudieras echarme una mano porque no tengo ni idea 
:(.

https://gyazo.com/176530f657b92772fd7fe283d5dc1c7c

Perdona las molestias y gracias !

**Reply by elpatronaco** [reply]
Primero de todo bajate la nueva versión, he añadido mas tiendas. Y ahí te dice que el login ha fallado, seguramente tengas el correo, o la contraseña mal. Saludos

**Reply by franVR10** [reply]
Gracias por contestar tan rápido! Esta es la última versión. Me la he descargado hará 1 hora o así. Ya que la versión antigua que tenía me decía que yo tenia chrome 89 y que mi version de chromedriver solo soportaba hasta la 87. Total, me descargué la versión 89, la instalé y ahora está así.
Los datos de inicio de sesión están perfectos y sí, le he mantenido la ' '. Sinceramente no entiendo que pasa. Me he descargado esta versión, la he cambiado al disco duro principal, he hecho de todo y sigo pensando que es algo de que no es capaz de pillar el chromedriver que debería...

**Reply by elpatronaco** [reply]
Digo la última última, la he subido hace media hora o algo así. Hazme caso vuelve a bajártelo, pones los datos (verifica que estén bien) y vuelves a probar a ver. Los datos ahora se ponen en un archivo que se llama **data.json**, lo verás

**Reply by franVR10** [reply]
Vale, me la acabo de descargar y he puesto de momento solo los datos de PcComponentes pero entiendo que me sigue fallando por algo:

https://gyazo.com/d0c98fecae643806d3ae986556451df7

**Reply by elpatronaco** [reply]
Vale eso es otro error, entiendo que solo has puesto los datos para pccomponentes no? En el archivo data.json, borra las entradas que dicen "coolmod" y "ldlc" y busca una entrada que se llama "debug" y ponla en true, inícialo de nuevo y dime si está entrando los datos correctamente y si inicia sesión

**Reply by franVR10** [reply]
He hecho lo que me has comentado y parece que va perfecto. Muchas gracias de verdad !

https://gyazo.com/b82e7ac3e0b89316d96848251afda60f

Por cierto, al abrirse chrome se crea una pagina en blanco, no se si es intencionado o no pero por si acaso aquí te lo dejo:

https://gyazo.com/c0409fdae1116ef3f8ced48aebb2df7c

**Reply by elpatronaco** [reply]
La página en blanco no es problema, aún asi en debug: true no te comprará en el momento en que haya stock, solo lo uso para ver que es lo que hace el bot, pero por código si debug es true no le dará al botón de comprar. Te tengo que mirar esto porque no te va. Crea un issue y pon la versión de chrome que tienes... Y cosas que puedan servir para reproducir el problema, y cuando tenga tiempo me lo miraré

**Reply by franVR10** [reply]
He vuelto a poner debug en false y no me ha salido ningún error. No se me ha abierto chrome pero el bot estaba actualizando el producto, debería ser así y funciona o va mal pero no me ha saltado el error?

**Reply by elpatronaco** [reply]
Pasa cap de los logs

**Reply by franVR10** [reply]
https://gyazo.com/5cfcbed5eddd65843bb75e67562b3721

Y por cierto 2 cosillas más:

-El bot una vez compra una para verdad?
-En esta última versión no hay para poner tarjeta? Tenemos que ponerla si o si en la página?

Ojo: Por lo que se ve solo me da error cuando pongo a coolmod en el data !

De nuevo, muchísimas gracias por todo!

**Reply by elpatronaco** [reply]
No, lo de que compre una y pare lo tengo que poner, y lo de la tarjeta no lo voy a poner, es trabajo que no lo veo necesario pudiendo hacer transferencia. Y otra cosa qué error te dice con coolmod? Intenta reproducirlo y pasame captura a ver si lo puedo arreglar

**Reply by franVR10** [reply]
Perfecto. Pues mira el error ese es el primero que te dije cuando me descargué esta nueva versión. Con LDLC no salta, ni con pccomponentes, pero si pones a coolmod en el data salta esto:

https://gyazo.com/d0c98fecae643806d3ae986556451df7

Y unas últimas cosillas:

Si no hay tarjeta puesta en pccomponentes que hace el bot? He leído antes creo que hace una transferencia pero no estoy seguro de como va eso porque siempre suelo pagar con paypal y tal, y tambien leí algo de que las guardas asi que me he quedado con la duda de como va jaja.

Por cierto sabes  como está la cosa con las 3080 últimamente? Salen en algún lado 1 aunque sea? Estoy suscrito en un canal de telegram que un bot supuestamente avisa cuando hay stock pero me parece que no acierta mucho la verdad... Me saca cada no mucho que sale una zotac trinity.

**Reply by elpatronaco** [reply]
El bot paga por transferencia, ya te digo es mucho trabajo el pagar con tarjeta, prefiero si en el futuro hay que añadir alguna tienda pues gastar ese tiempo en escribir el script para la tienda y no para añadir tarjetas. El error ese voy a a hacer un commit ahora solucionándolo, y sobre las 3080 ni idea hahaha yo ahora voy a por la rx6700xt que sacan el 18 de este mes creo, y estoy buscando en qué tiendas saldrá pero d momento no he encontrado nada

**Reply by elpatronaco** [reply]
Ya he hecho los cambios, no te debería fallar, y en el data.json he metido una opción que es __onlyOneBuy__ en la que cuando compre una gráfica se cierra el script

**Reply by franVR10** [reply]
Si si me parece perfecto que sea por transferencia pero me refiero... Si el bot no te pide en ningún momento datos para el pago de transferencia como va? Se queda la orden lo que compres guardada hasta que confirmes el pago? Es que nunca se me ha dado la casualidad de pagar por transferencia jeje. Voy a probar ahora a ver que tal con los cambios. Y muchisisimas gracias por todo tio, me sabe mal no darte nada a cambio del trabajazo que te has pegado en general !

EDIT IMP: 
Coolmod sigue dándome el único fallo :  https://gyazo.com/72d80edaf16d041499fc8d0ebb9fdf45

**Reply by elpatronaco** [reply]
En general en todas las tiendas se quedan los pedidos esperando a q hagas el pago, al momento de la compra no te piden ningun dato. Y no me tienes que dar nada hahaha tranquilo, ya me han dicho varias personas que dejara la dirección de bitcoin o el correo de paypal para donarme y he dicho que no, no me dedico a esto xd. Simplemente disfrútalo y si al final te compras la gráfica ponlo por aquí que hace ilusión ayudar a la gente. Saludos loco :)

**Reply by elpatronaco** [reply]
Y lo del error pff, me lo miraré cuando pueda, pensaba q lo habia arreglado pero parece q no

**Reply by franVR10** [reply]
Oohh pues perfecto la verdad. Yo si te habrieras alguna de esas te daba algo 100%, muchas gracias de nuevo :). Si me la puedo comprar, madre mía, te amaré lo más grande jaja. Me parece muy interesante estos programillas, yo estoy en 2º de Ingeniería Informática y poco a poco ya nos van enseñando cositas útiles.

Muchas gracias de nuevo por el esfuerzo!

**Reply by elpatronaco** [reply]
Yo estoy en segundo de grado superior de DAM, el año que viene hago ingeniería informática también. No habéis dado programación en la carrera aún? En el grado en segundo entre la dual y las clases yo ya sé python, el stack MERN, C# y apps windows WPF... Joder ahora me da miedo que en la uni me expliquen más física que programación

**Reply by franVR10** [reply]
Pues en primero casi todo es matemáticas pero entre 1º y 2º nos enseñan: C++, java , esamblador, C, haskell, R, SQL(DML y DDL). Y muchas técnicas de programación principalmente en java.
1º es el año más complicado la verdad...  Lo demás como se coge con ganas no tanto :)

**Comment by TriumphJ** [comment]
Me da este fallo al ejecutar el start
TypeError: page.evaluate(...).includes is not a function

Aun asi sigue corriendo, listando las graficas y diciendo que no estan disponibles, pero no se porque da ese error mientras se ejecuta

**Reply by elpatronaco** [reply]
Qué tienda?

**Reply by TriumphJ** [reply]
Tengo el inicio de sesión en las 3 tiendas, pcc colmood y lcld
El tema es que por lo que veo si funciona, (aunque no me abre ninguna ventana, no se si lo hace por detras)

PD: me han enviado un correo de colmood diciendo que envio muchas peticiones a la web, y que es baneable, supongo que habria que reducir el 1000 a otro numero

**Comment by yopiiiii** [comment]
Hola, acabo de instalarme el script y por fin he conseguido hacerlo funcionar, muchas gracias por el curro elpatronaco.

Ahora tengo algunas dudas:

-¿Se pueden poner varias a la vez de pccomponentes con el mismo script? ¿O tengo que copiarme todo el sistema de ficheros para cada instancia que ejecute?
-¿Con las que son reacondicionadas no funciona el script, verdad? ¿Hay alguna forma de hacer que funcione?
-Y sobre todo, no se que gráficas tienen probabilidades de que vuelva a haber stock, hay millones de variaciones de cada modelo. ¿Como sabeis que tarjetas poner en cada momento? Me interesa la serie 20XX (2070super hasta 2080 super sobre todo) más que la 30 (aunque si sólo hay de estas, pues palante), y algo que no se vaya mucho de precio. ¿Ahora mismo cuales pondríais? ¿Podeis pasar enlaces?

Muchas gracias de nuevo.

**Comment by alfredopg11** [comment]
Buenas.

En primer lugar, felicidades por el trabajo. El refreshRate en qué unidades está? Por más que lo cambio, a muy pequeño o a muy grande, en la consola se sigue actualizando con la misma frecuencia.

Saludos.

**Reply by elpatronaco** [reply]
Milisegundos

**Comment by DarthWendigo** [comment]
Buenas, gracias por el trabajo @elpatronaco , a ver si podemos arrebatar algo a los mineros.

Al hacer el npm start me salta este error, me he hecho pull de la última versión:

![image](https://user-images.githubusercontent.com/2934132/111924042-b91db900-8aa2-11eb-93dd-1c34d2faa2c5.png)

Gracias por la ayuda y un saludo.

**Reply by elpatronaco** [reply]
Buenas, fa pinta que tienes una versión de Node vieja, las funciones asíncronas están soportadas a partir de Node 7.6. Míralo con el comando **node -v** y me dices. Pero no es un error del script

**Reply by DarthWendigo** [reply]
Estaba utilizando la 10.16, aún así la he subido a la última, 14.16 y ahora me sale esto, he verificado la contraseña de pccomponentes y está ok.
![image](https://user-images.githubusercontent.com/2934132/111973791-27906480-8aff-11eb-839b-6459ee50fc4c.png)
Gracias!

**Reply by RodrigoTomeES** [reply]
@DarthWendigo actualiza el bot, @elpatronaco ha arreglado ese error en un nuevo commit, lo comenta en este issue #18

**Reply by DarthWendigo** [reply]
Ahora sí gracias, me extrañaba porque hice pull anoche, pero ya he visto que hay commits nuevos.

**Comment by DarthWendigo** [comment]
Buenas, he metido que saque cuando detecta stock el precio al que lo detecta pero el "resp" parece que viene vacío siempre, ¿es normal?

![image](https://user-images.githubusercontent.com/2934132/112065812-a0c2a280-8b65-11eb-9824-f2d53d7d87c1.png)

Edito: He visto que en la captura que os pongo ahora, en la línea 88 resp.price sí tiene valor, pero en la línea 90 no, siempre 0. Perdonad pero es muy probable que se me escape algo de Node que he hecho muy poquito, pero por entenderlo ^^.

![image](https://user-images.githubusercontent.com/2934132/112074412-834a0480-8b76-11eb-8d61-68d754ad1764.png)

Un saludo.

**Comment by RodrigoTomeES** [comment]
Hola buenas,

Hay alguna manera de ver los mensajes del bot si has cerrado la sesión ssh? Es que ha salido una 3070 reaco en pccomponentes y no la ha pillado supongo que habrá sido un bot más rápido pero no se... por ver si intentó pillarla

**Reply by DarthWendigo** [reply]
Me ha pasado igual, sería interesante enviar los mensajes cuando detecta stock a disco quizás

**Reply by elpatronaco** [reply]
no sería demasiado difícil poner avisos x telegram, cuando pueda lo miro, pero guardar los logs en el disco es un puto coñazo. No por programarlo eh, si no por ir a la máquina cada vez que quieres ver los logs

**Reply by elpatronaco** [reply]
podéis mirar la nueva rama ? Sólo envia mensajes, lo he hecho mientras me acaba de compilar otra cosa del curro

**Reply by elpatronaco** [reply]
Si quieres meter comandos y historias todo tuyo @RodrigoTomeES, se podría hacer un comando para empezar a buscar una grafica por ejemplo, yo que sé

**Reply by RodrigoTomeES** [reply]
Okay gracias, a ver si en un rato lo miro 👌

**Reply by RodrigoTomeES** [reply]
Ha salido una 3070 a 500 en pccomponentes y nada, lo peor es que he mirado los logs y no ha detectado el stock que raro. Tú @DarthWendigo has tenido suerte?

Actualizo, otras dos más y nada, que raro joe

**Reply by DarthWendigo** [reply]
Tampoco, lo había parado justo además para actualizar.. qué enlace era? yo tengo este metido:
https://www.pccomponentes.com/gigabyte-geforce-rtx-3070-gaming-oc-8gb-gddr6-reacondicionado

**Reply by RodrigoTomeES** [reply]
Si ese tengo yo también, y han salido varias vision y nada no entiedno joder que asquete

**Comment by DarthWendigo** [comment]
Buenas, alguno ha probado a meterlo en un Lambda? Había pensado en subirlo a un Lambda y sacar luego los logs a Cloudwatch por ejemplo.

**Reply by elpatronaco** [reply]
Creo que te vendría mejor una instancia d S3 que no un lambda. Primero que el programa no está diseñado con las lambdas en mente y no creo q funcione a menos que lo refactorices. Y creo q tendrías problemas de rendimiento

**Reply by DarthWendigo** [reply]
Habría que hacer cambios sí (el handler hay que definirlo y hacerle export por lo que he leído), ahora mismo tengo en Lambda varias cosas del curro para automatizarlas en Python, pero sí es cierto que en Node se me escapa un poco y tardaría más en dar con la tecla. Lo más sencillo sería un EC2 más que un bucket de S3.

**Reply by elpatronaco** [reply]
Ec2 si hahaha, que yo no controlo mucho de aws y se me mezclan los nombres, o un contenedor en ecs, con el dockerfile que hay puedes generar una imagen. Yo en el curro he desarrollado apis en nodejs diseñadas para lambdas, creo que es para una de las unicas cosas que sirven las lambdas, más allá de eso...

**Comment by DarthWendigo** [comment]
He visto que de vez en cuando, después de salir que ha detectado stock en coolmod, sale esto unas líneas después:

TimeoutError: waiting for selector `button[class='button-buy']` failed: timeout 30000ms exceeded

Supongo que será que porque se agota rápidamente y ya no está ese botón, ¿correcto?

Gracias!

**Reply by RodrigoTomeES** [reply]
Yo he visto que en la última ejecución de los test sobre node 12 y 15 lanza un error tambíen el logging de Coolmod, este de aquí:

![imagen](https://user-images.githubusercontent.com/13077343/112480396-48162400-8d76-11eb-891d-3d742feea03f.png)

Por cierto, @elpatronaco creo que están mal los credenciales que hay en el repo de LDLC
![imagen](https://user-images.githubusercontent.com/13077343/112480644-8875a200-8d76-11eb-9204-006ea676c548.png)

**Reply by DarthWendigo** [reply]
Guay las últimas mejoras de log, pero el precio cuando hay stock ocurre lo que os comentaba, que el resp viene vacío por lo que el texto no sale y el precio sale 0. Si le preguntamos antes de if stock, sí tiene la info.

[coolmod] : PRODUCT IN STOCK at 0€! Max price set is 650€.  Starting buy process

Voy a indagar un poco más.
